modules/SRanipal_module/src/GazeRetargeting.cpp

```diff
@@ -441,14 +441,41 @@ iDynTree::Vector2 GazeRetargeting::VRInterface::applyDeadzone(const iDynTree::Ve
     Eigen::Map<Eigen::Vector2d> outputMap = iDynTree::toEigen(output);
 
     double inputNorm = map.norm();
-    if (inputNorm > m_errorDeadzone)
+    bool deadzoneNotActiveAndErrorStillHigh = !m_deadzoneActive && inputNorm > m_errorDeadzone;
+    bool deadzoneActiveButErrorVeryHigh = m_deadzoneActive && inputNorm > m_errorDeadzoneActivation;
+    bool timeCheckNotActive = m_deadzoneMinActivationTimeInS <= 0.0;
+    bool timeCheckPerformedOnce = m_deadzoneActivationTime >= 0.0;
+    bool enoughTimePassed =  timeCheckNotActive || (timeCheckPerformedOnce && ((yarp::os::Time::now() - m_deadzoneActivationTime) >= m_deadzoneMinActivationTimeInS));
```

Comment:
@S-Dafarra  can you explain a bit of strategy you took for these lines of code?

Reply:
What you did makes sense. But it is still hard to follow the logic here even with the meaningful variable names, at least for me. So, that's why I asked you to drop some description of the logic here.

Reply:
Sorry, I think I did not understand. Can you be more specific on the logic you are referring to?

Reply:
I am asking you to explain those lines!